lib/AppInfo/Application.php

```diff
@@ -200,7 +200,9 @@ public function updateCSP() {
 
 		// Do not apply CSP rules on WebDAV/OCS
 		// Ideally this could be a middleware running after the controller execution before rendering the result to only do it on page response
-		if ($container->getServer()->getRequest()->getScriptName() !== '/index.php') {
+		$script_name_parts = explode('/', $container->getServer()->getRequest()->getScriptName());
+		$script_file = end($script_name_parts);
+		if ($script_file !== 'index.php') {
```

Comment:
```suggestion
		$scriptNameParts = explode('/', $container->getServer()->getRequest()->getScriptName());
		$scriptFile = end($scriptNameParts);
		if ($scriptFile !== 'index.php') {
```

To stick to https://docs.nextcloud.com/server/latest/developer_manual/getting_started/codingguidelines.html#objects-functions-arrays-variables 😉